Backend/Backend/Services/EpisodeService.cs

```diff
@@ -22,12 +28,32 @@ public class EpisodeService : IEpisodeService
 
         }
 
+        public double GetEpisodeDuration(string path)
+        {
+            try
+            {
+                Console.WriteLine(path);
+                using (var audioFileReader = new AudioFileReader(path))
+                {
+                    Console.WriteLine(audioFileReader.TotalTime);
+                    return audioFileReader.TotalTime.TotalSeconds;
+                }
+            }
+            catch (Exception ex)
+            {
+                
+                Console.WriteLine("Error: " + ex.Message);
+                return TimeSpan.Zero.TotalSeconds;
+            }
+
+        } 
+
 
         public async Task<Episode?> AddEpisode(CreateEpisodeRequest createEpisodeRequest,Podcast podcast,HttpContext httpContext)
         {
             Episode episode = new Episode();
 
-            if(!AllowedTypes.Contains(createEpisodeRequest.AudioFile!.ContentType))
+            if(createEpisodeRequest.AudioFile != null && !AllowedTypes.Contains(createEpisodeRequest.AudioFile!.ContentType))
```

Comment:
you have this backwards, if It IS null, we want to throw the error here, not if it isnt

Reply:
you can disregard this. you are correct here. Add a comment however to show it, since the logic is kinda confusing